tools/purge_omsagent.sh

```diff
@@ -0,0 +1,407 @@
+    fi
+    if [ $? -ne 0 ]; then
+        echo "----- Ignore previous errors for removing group: $1 -----"
+    fi
+}
+
+# $1 - The name of the directory to delete (if it exists)
+# $2 - The name of the package associated with the directory
+dir_rm()
+{
+    if [ -d $1 ]; then
+        pkg_rm $2
+        if [ -d $1 ]; then
+            rm -rf $1
+        fi
+    fi
+}
+
+
+
+##### PURGE SCRIPT FOLLOWS #####
+
+
+
+ulinux_detect_installer
+ulinux_detect_servicecontroller
+check_waagent_name
+
+# REMOVE LAD
+check_if_pkg_is_installed lad-mdsd
+LAD_MDSD_INSTALLED=$?
+if [ $LAD_MDSD_INSTALLED -eq 0 ]; then
+    echo "--------------------------------------------------------------------------------"
+    echo "WARNING: LAD is currently installed on the machine."
+    echo "  Purging OMS without purging LAD may not fully succeed, because LAD has a dependency on OMS."
+    echo "  In order to successfully purge OMS, LAD must be purged as well."
+    while : ; do
+        read -p "Do you wish to continue with purging? (y/n)" continue_lad
+        if [ "$continue_lad" = "y" ]; then
+            echo "Continuing with purge..."
+            echo "--------------------------------------------------------------------------------"
+            break
+        elif [ "$continue_lad" = "n" ]; then
+            echo "Exiting without purging..."
+            call_exit 0
+        else
+            echo "Unknown input. Please type 'y' (yes) or 'n' (no) to proceed."
+        fi
+    done
+
+    # continuing with LAD purge
+    echo "Please go to portal.azure.com and uninstall the LAD extension."
+    echo "Go to Azure Portal -> Virtual Machines -> <vm_name> -> Settings -> Extensions"
+    echo "and then click the '...' in the LinuxDiagnostic row and click 'Uninstall'"
+    read -p "Press enter to proceed once the Extension is uninstalled." toss2
+    echo ""
+
+    # remove LAD directories
+    echo "---------- Removing LAD directories ----------"
+    if [ -d $SOLO_LAD_DIR ]; then
+        rm -rf $SOLO_LAD_DIR
+    fi
+    if [ -d $OMS_LAD_DIR ]; then
+        rm -rf $OMS_LAD_DIR
+    fi
+
+    # remove LAD package
+    echo "---------- Removing LAD package ----------"
+    pkg_rm lad-mdsd
+
+    # remove MDSD package
+    check_if_pkg_is_installed mdsd
+    MDSD_INSTALLED=$?
+    if [ $MDSD_INSTALLED -eq 0 ]; then
+        echo "---------- Removing MDSD package ----------"
+        pkg_rm mdsd
+    fi
+
+    echo "LAD successfully purged!"
+fi
+
+
+# REMOVE OMS
+echo "--------------------------------------------------------------------------------"
+echo "------------------------------ STARTING OMS PURGE ------------------------------"
+echo "--------------------------------------------------------------------------------"
+echo ""
+
+# Check if installed via extension
+ls /var/lib/waagent | grep -i Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux- > /dev/null 2>&1
+EXTENSION_INSTALLED=$?
+if [ $EXTENSION_INSTALLED -eq 0 ]; then
+    echo ""
```

Comment:
If the customer has a policy or ASC set up to push OMS agent to the machine, that will end up pretty quickly re-installing the OMS extension; maybe that should be called out here. It might not be a big concern as long as they come back to the purge script and quickly resume, but that is not a given ...

Reply:
Fixed in this commit: https://github.com/microsoft/OMS-Agent-for-Linux/pull/1255/commits/5849d518e8127291f13356832c8c7a023e24041e